commit d2878984a8cf80054ad3298431bce644643addf3
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Wed Oct 11 13:48:04 2017 +0100

    amba-pl011: Report AUTOCTS capability to framework
    
    The PL011 has full hardware RTS/CTS support which is enabled by
    the driver when flow control is requested. However, it doesn't
    notify the UART framework of the fact, causing the software CTS
    support to be enabled at the same time.
    
    Software CTS triggers the sending of another batch of characters
    when CTS becomes asserted. The pl011 interrupt handler processes
    the CTIS bit before TXIS, which can cause some characters to be
    sent between the time that the TXIS bit first becomes asserted
    and the time it is handled by a call to px011_tx_chars. This
    would be fine were it not for the optimisation in pl011_tx_char
    that assumes the FIFO is half-empty if called from the interrupt
    handler and skips the checking of the FIFO status register before
    sending each character, leading to data loss if the FIFO is more
    than half-full.
    
    Prevent the data loss and improve efficiency by indicating the
    AUTOCTS support.
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/drivers/tty/serial/amba-pl011.c b/drivers/tty/serial/amba-pl011.c
index b224306633f6..3b743cf8a94c 100644
--- a/drivers/tty/serial/amba-pl011.c
+++ b/drivers/tty/serial/amba-pl011.c
@@ -2047,9 +2047,11 @@ pl011_set_termios(struct uart_port *port, struct ktermios *termios,
 
 		old_cr |= UART011_CR_CTSEN;
 		uap->autorts = true;
+		port->status |= UPSTAT_AUTOCTS;
 	} else {
 		old_cr &= ~(UART011_CR_CTSEN | UART011_CR_RTSEN);
 		uap->autorts = false;
+		port->status &= ~UPSTAT_AUTOCTS;
 	}
 
 	if (uap->vendor->oversampling) {
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1131_d2878984a8cf80054ad3298431bce644643addf3.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1131_d2878984a8cf80054ad3298431bce644643addf3.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
